src/server/svr_comm.c

```diff
@@ -257,29 +259,30 @@ replyhello_psvr(struct work_task *ptask)
  * stats only a few attributes. This request is asynchronous
  * and server will process when it gets a PS_STAT_RPLY.
  * 
+ * @param[in] type - msvr stat type
  */
 void
-send_nodestat_req(void)
+send_nodestat_req(enum msvr_stats_type type)
 {
 	struct attrl *pat;
 	struct attrl *head;
 	struct attrl *tail;
 	struct attrl *nxt;
 	int mtfd = open_ps_mtfd();
-	int rc;
 	static int time_last_sent = 0;
+	int rc;
 
-	update_msvr_stat(1, CACHE_MISS);
+	update_msvr_stat(1, type);
```

Comment:
All the calls of send_nodestat_req() take CACHE_MISS, so, why did you change this to access CACHE_MISS through `type` variable?

Reply:
type can also be -1. Which is handled within update_msvr_stat instead of the caller.